### PR TITLE
Fixing #16 and introducing polyfill namespaces.

### PR DESCRIPTION
**Pull request by thirtytwobits**

### CODE REVIEW

File: include/cetl/cetl.hpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+/// @{
+
+/// @def CetlCppStandard14
+/// Provides the proper value to test against `__cplusplus` for c++14.
+/// ```
+/// #if __cplusplus >= CetlCppStandard14
+/// #include <something_from_cpp_14>
+/// #endif
+/// ```
+#define CetlCppStandard14 201402L
```
Comment (pavel-kirienko) [inline include/cetl/cetl.hpp:57]: Maybe I missed something, why did you change these from the screaming snake case?

**Review by pavel-kirienko — APPROVED** [review]

File: include/cetl/pf20/cetlpf.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+/// ```
+///
+
+#ifndef CETL_PF20_H_INCLUDED
+#define CETL_PF20_H_INCLUDED
+
+#include "cetl/cetl.hpp"
+
+#if (__cplusplus >= CetlCppStandard20)
+#    include <span>
```
Comment (emrainey) [inline include/cetl/pf20/cetlpf.hpp:42]: Is this std:: style?

File: include/cetl/pf20/cetlpf.hpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+#else
+#    include "cetl/pf20/span.hpp"
```
Comment (emrainey) [inline include/cetl/pf20/cetlpf.hpp:45]: here too

File: include/cetl/cetl.hpp (hunk 1 above)
Comment (emrainey) [reply include/cetl/cetl.hpp:57]: I'm confused about that too. I dislike the macros (they are avoidable but with some hassle) but if they remain, should be as they were..

Comment (thirtytwobits) [reply include/cetl/cetl.hpp:57]: Because, if we were to drop C++14 these would become inline variables. This future proofs the tokens for such a glorious future (which is not forthcoming this year nor next but perhaps someday).

File: include/cetl/pf20/cetlpf.hpp (hunk 2 above)
Comment (thirtytwobits) [reply include/cetl/pf20/cetlpf.hpp:42]: Yeah. @pavel-kirienko , I'm getting the right clang-format style here right?

Comment (pavel-kirienko) [reply include/cetl/pf20/cetlpf.hpp:42]: Seems so.

File: include/cetl/cetl.hpp (hunk 1 above)
Comment (pavel-kirienko) [reply include/cetl/cetl.hpp:57]: >Because, if we were to drop C++14 these would become inline variables.

Wait what? Are you saying that in C++17+ you can expand inline variables in preprocessor conditions?

Comment (thirtytwobits) [reply include/cetl/cetl.hpp:57]: Indeed: https://godbolt.org/z/c3v3Y8YT7

Comment (thirtytwobits) [reply include/cetl/cetl.hpp:57]: Yeah, I'm an idiot. I'll `CHANGE_THESE_BACK_SORRY`.

**Review by pavel-kirienko — APPROVED** [review]

**Review by emrainey — APPROVED** [review]
